### ISSUE TITLE
xCapture FSEQs out-of-sync when paired with xLight Sequences

### ISSUE THREAD

**Issue by jholmes09** [smeighan/xLights#1294]
I am running xLights 2018.34 on Windows 10.
I am also using GrandMA2 onPC software.

When I use xSchedule as a Midi Timecode Master, timecode is sent to the GrandMA onPC software and lighting cues are triggered perfectly with the music.

I am using xCapture to encapsulate one universe of SACN output from GrandMA onPC, which outputs at 33/34ms.
Despite verifying the refresh rate via wireshark, xCapture insists the refresh is 30ms.

When I bring the captured FSEQ into xSchedule and play against the same audio file, or if I import the FSEQ into an xLights sequence file, the playback of the capture drifts over time, often starting slightly ahead or behind but ultimately ending up a good 6-7 seconds behind the music & other sequence data (at roughly 60-70 seconds into the song).

I have attempted to start/stop the capturing using a DMX trigger and a mouse-click macro, but that has not helped.

I have adjusted my sequences to match the refresh-rate. I have tried 30ms and 33ms with no change.

I have also tried changing my SACN output to 127.0.0.1 instead of Multicast onto the actual network.

Not really sure what else to try. I have attached a sequence and multiple captured FSEQs.
Be advised the audio file has LTC Timecode on the left channel so it will SCREECH!
Mute the left channel of your audio output before you play.

[GrandMA2 Captures.zip](https://github.com/smeighan/xLights/files/2504940/GrandMA2.Captures.zip)

[grandma_timecode_test_v7.zip](https://github.com/smeighan/xLights/files/2504964/grandma_timecode_test_v7.zip)

**Comment by keithsw1111** [comment]
The frame rates need to match. 

You can tell capture not to guess the frame rate and specify it yourself.

**Comment by jholmes09** [comment]
@keithsw1111 I did. 

Even with both the xLights sequence and the captured FSEQ at 33ms, the captured file drifts.

**Comment by keithsw1111** [comment]
You have to build both with the same frame rate and when you capture it you have to set the right frame rate. If you get all of them right they should hold time ... but you have to get them right.

If the grand ma sends at 33 but you say it is 34 then it will run slightly slow.

**Comment by jholmes09** [comment]
Is there a way to change the rate within the FSEQ after it is captured? xCapture only gives me a few options for frame rates and I have compared what xCapture thinks the rate should be vs the wire shark output and they’re not the same. 

To that end though, I’ve been doing as you said, making sure both the capture and sequence are the exact same frame rate. Yes I still get an ever increasing drift.

**Comment by keithsw1111** [comment]
With a hex editor you can. but not really practical.

I could make it so that you can set any frame rate.

As for the drift I dont see how that is possible unless the framerate is not an exact match. remember there are two ways to read a frame rate fps and ms. We use ms. Not all fps ratings translate accurately into ms. If your board is using fps then that could explain the drift.

**Comment by jholmes09** [comment]
You are correct. Any discussion with MA Lighting has them saying "FPS" not MS.
To be sure, I've been sampling the capture using WireShark and looking at the refresh rate from the GrandMA. That is where I confirmed the 33ms refresh. 

If making the frame-rate user selectable is not difficult, I'll happy test it out.
I think I should be able to do a single capture and then hit the save button multiple times to save at different rates for testing purposes. 

The only other thought I would have is that this is part of the "animation FSEQs play too fast" issue I reported before. I've been having trouble quantifying the data on that though so I'm looking to find tangible facts before I proceed down that trail.
